src/components/shared/Fields/Source.js

```diff
@@ -25,9 +26,9 @@ class SourceField extends PureComponent {
     source: null,
   };
 
-  handleSelect = (source) => {
-    this.props.onChange(this.props.getSourceById(source.id));
-  };
+  handleSelect = (source) => this.props.onChange(this.props.getSourceById(source.id));
+
+  handleRemove = () => this.props.onChange({ id: null });
```

Comment:
pass a simple null to `onChange` not a crippled object. Let the caller handle it.

Reply:
done